sklearn/preprocessing/_encoders.py

```diff
@@ -70,6 +71,14 @@ def _get_feature(self, X, feature_idx):
         # numpy arrays, sparse arrays
         return X[:, feature_idx]
 
+    def _check_pandas_categories(self, cats, pd_category):
+        """Checks cats is lexicographic consistent with categories in fitted X.
+        """
+        msg = ("'auto' categories is used, but the Categorical dtype provided "
+               "is not consistent with the automatic lexicographic ordering")
```

Comment:
What should the user do?

Reply:
Is it worth listing or diffing the category list?

Reply:
> What should the user do?

Right now...nothing good. The user would need order their categorical dtype to be lexicographic ordering to avoid this warning, which I think is bad. This should most likely be a `FutureWarning` suggesting that in the future we will respect the Categorical dtype. We can add support for a `categories='dtype'` which will respect the Categorical dtype.

> Is it worth listing or diffing the category list?

Sometimes the only difference is the order. What kind of diff would be good to have?

Reply:
Done. Also added another sentence regarding passing a list to the categories parameter.